### PR TITLE
Update documentation of `upwind.Upwind`

### PR DESCRIPTION
**Pull request by pschultzendorff**
## Proposed changes
- Addresses #1237 by updating the outdated documentation of the `upwind.Upwind` class. This includes correcting parameters, return values, and required entries in data dictionaries, following the style of `tpsa.Tpsa`.
- Fixes inline comments that became obsolete after #1192.
- Resolves minor style issues I found in the process.

## Questions
- Does this lift the documentation to a level we are happy with? I fixed problems I found while writing a two-phase flow model, but there might be more that I missed.
- Shall I update the documentation for `upwind.UpwindCoupling` as well? This would bring the entire module up to date, isn't too much work, and I would gladly fix it.

## Types of changes
- [ ] Minor change (e.g., dependency bumps, broken links).
- [ ] Bugfix (non-breaking change which fixes an issue).
- [ ] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).
- [ ] Testing (contribution related to testing of existing or new functionality).
- [x] Documentation (contribution related to adding, improving, or fixing documentation).
- [ ] Maintenance (e.g., improve logic and performance, remove obsolete code).
- [ ] Other:

## Checklist
- [ ] The documentation is up-to-date.
- [x] Static typing is included in the update.
- [x] This PR does not duplicate existing functionality.
- [x] The update is covered by the test suite (including tests added in the PR).
- [ ] If new skipped tests have been introduced in this PR, `pytest` was run with the `--run-skipped` flag.

### CODE REVIEW

File: src/porepy/numerics/fv/upwind.py [hunk 1]
```diff
@@ -123,38 +158,45 @@ def discretize(self, sd: pp.Grid, data: dict) -> None:
...
+                Stored in ``data[pp.DISCRETIZATION_MATRICES][self.keyword]``
+
+        parameter_dictionary contains the entries:
+            - bc: :class:`~porepy.params.BoundaryCondition`.
+                Boundary conditions for the advected property.
+            - darcy_flux: :class:`~numpy.ndarray` of ``shape=(sd.num_faces,)``
+                Normal velocity at each face, weighted by the face area.
+                Values of the boundary conditions. The dictionary has at most the
+                following keys: 'dir' and 'neu', for Dirichlet and Neumann boundary
+                conditions, respectively.
```
[the remark this answers is not in the document]
Comment (pschultzendorff) [reply src/porepy/numerics/fv/upwind.py:178]: I think I forgot to delete something here.

File: src/porepy/numerics/fv/upwind.py [hunk 2]
```diff
@@ -240,7 +281,7 @@ def discretize(self, sd: pp.Grid, data: dict) -> None:
         # caution. The below implementation follows the following steps:
         #
         # 1) On Neumann boundaries the prescribed boundary value should effectively be
-        # added to the adjacent cell, with the convention that influx (so negative
+        # added to the adjacent cell, with the convention that influx (so positive
```
[the remark this answers is not in the document]
Comment (pschultzendorff) [reply src/porepy/numerics/fv/upwind.py:289]: Thanks for the catch!

**Comment by pschultzendorff** [comment]
Before I merge, what about the `UpwindCoupling` discretization? As said, it wouldn't be an issue to quickly fix the class as well and then the whole module is dealt with.

**Comment by IvarStefansson** [comment]
> Before I merge, what about the `UpwindCoupling` discretization? As said, it wouldn't be an issue to quickly fix the class as well and then the whole module is dealt with.

Please do as you please. Leaving it for later is completely fine. If you do want to go for it, please don't spend too much time, particularly for assemble_matrix_rhs, which is only kept for legacy reasons.

File: src/porepy/numerics/fv/upwind.py [hunk 3]
```diff
@@ -456,33 +507,66 @@ def assemble_matrix_rhs(
...
+                ``shape=(sd.num_faces, sd.num_cells)``. Upwind matrix for the
+                primary domain.
+            - ``self.upwind_secondary_matrix_key``: :class:`~scipy.sparse.csr_matrix`
+                of ``shape=(sd.num_faces, sd.num_faces)``. Upwind matrix for the
+                secondary domain.
+            - ``self.flux_matrix_key``: :class:`~scipy.sparse.csr_matrix`
+                of ``shape=(sd.num_faces, sd.num_faces)``. Flux matrix.
+            - ``self.mortar_discr_matrix_key``: :class:`~scipy.sparse.csr_matrix`
+                of ``shape=(intf.num_cells, intf.num_cells)``. Identity matrix for the
+                mortar variable.
```
Comment (IvarStefansson) [inline src/porepy/numerics/fv/upwind.py:551]: There is quite a bit of duplicate documentation in this file now. Sometimes, we prefer referring to docstrings of other function. You may want to consider this in the future (not this file, though).

**Review by IvarStefansson — APPROVED** [review]
Thanks for your effort!

File: src/porepy/numerics/fv/upwind.py (hunk 3 above)
Comment (pschultzendorff) [reply src/porepy/numerics/fv/upwind.py:551]: Will keep this in mind in the future!

**Pull request merged**